app/models/hotel.rb

```diff
@@ -1,3 +1,9 @@
 class Hotel < ApplicationRecord
   has_many :rooms
+
+  def list_of_guests
+    Guest.joins(:guest_rooms)
+    .where(guest_rooms: { room: rooms })
+    .distinct
```

Comment:
This query works, nice job! 
One way to make this easier would be to create another association from Hotel to Guests. But how would we do that? 

```
has_many :guests, through: :rooms
```
^ That works because of the association on `Room` of `has_many :guests, through: :guest_rooms`. Therefore it logically follows that a Hotel, which has many rooms, would also have many guests through those rooms. 

Try adding this in for practice, and adjusting your query to that new association. What would that look like?